cypher-shell/src/main/java/org/neo4j/shell/cli/InteractiveShellRunner.java

```diff
@@ -168,13 +170,23 @@ AnsiFormattedText getPrompt() {
         //  but that does not work in general)
         databaseName = ABSENT_DB_NAME.equals(databaseName) ? DEFAULT_DEFAULT_DB_NAME : databaseName;
 
-        AnsiFormattedText prompt = AnsiFormattedText.s().bold()
+        int promptIndent = connectionConfig.username().length() + 1 + databaseName.length() + 2;
+        AnsiFormattedText prePrompt = AnsiFormattedText.s().bold()
                 .append(connectionConfig.username())
                 .append("@")
-                .append(databaseName)
-//                .appendNewLine()
-                .append( txHandler.isTransactionOpen() ? TRANSACTION_PROMPT : FRESH_PROMPT );
-        return prompt;
+                .append(databaseName);
+
+        if (promptIndent <= 50) {
```

Comment:
Why just this number? And maybe make this a constant.

Reply:
It was deemed big enough to allow one-line prompts for common cases, but small enough not to waste all the width for really narrow terminals (80 chars). Made a constant.